index.html

```diff
@@ -1683,6 +1683,14 @@ <h3>Key types and formats</h3>
 Bitcoin format [[BASE58]] using the <code>publicKeyBase58</code> property.
               </td>
             </tr>
+            <tr>
+              <td>
+JWK<br/>(<code>JsonWebKey2020</code>)
```

Comment:
To be clear, I still object to key types where you cannot tell if the key was meant to have public key material or private key material. We have special rules in DID Core that help you understand that, but take this key type of this environment and it turns into a foot gun where processors can't determine whether it was meant to hold private key material or public key material.

Reply:
It's not that JOSE developers don't understand your argument. It's that they find being able to work with existing software more valuable than absolute conformance to your aesthetic preferences.

Web Crypto happily understands the difference between JWKs.

Web Crypto has no idea what `Ed25519VerificationKey2018` and `publicKeyBase58` are.

Blocking support for legacy (in your mind, perhaps misguided) key representations is not the same as designing / innovating newer safer representations.

Reply:
> Web Crypto has no idea what Ed25519VerificationKey2018 and publicKeyBase58 are.

Web Crypto has no idea what `JsonWebKey2020` is either... :)

You're inventing a new type that's broken in a way that has long been identified. You will understand the issue when people use the JsonWebKey2020 foot gun outside of DID Documents to accidentally publish private key material. 

> Web Crypto happily understands the difference between JWKs.

Can you explain how Web Crypto knows the intent behind a key expression? How does it know if the developer intended to express a public key vs. a private key?

Reply:
https://mdn.github.io/dom-examples/web-crypto/export-key/index.html

``
{
 "crv": "P-384",
 "d": "JbPKurXh-rPefsLGZJ3c4wQNQ9nKKtNTvwQFfQBpnW1-oq4OTzzAKGXG0rAKbf7o",
 "ext": true,
 "key_ops": [
  "sign"
 ],
 "kty": "EC",
 "x": "rGpcD0j9TxQ90J7KnIehMq9G-rhmMjkbk1g3CI_wy_jkxCfHPtyU5PQ_15Uaz5IH",
 "y": "z3QT9Rzf4kFS2TZhxkGskThAN_wKhdaYwg2TfxNDhrbHLAnAEFBPMOZj3Qz-oMYl"
}
```

Web Crypto is substantially more documented than linked data crypto. 

https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API

In general, no software knows what a developer meant to do when they accidentally publish a `publicKeyJwk` with `d` or an `Ed25519VerificationKey2018` with `privateKeyBase58`.... there is no solution for human error, but clear documentation can help mitigate it.

Reply:
> In general, no software knows what a developer meant to do when they accidentally publish a publicKeyJwk with d 

Why not? You have all the information you need, don't you? It's meant to be a public key -- it says it right there. Why are you not throwing an error when you see `d` in a public key!? :)

Note that the important information came from *outside* the JOSE structure... actually looking at the structure tells you nothing... you just assume it was always meant to be a private key... but it's both a public and a private key -- which is the whole point (and one of the problems with JOSE).

Also note that if the external structure said `privateKeyJwk`, you still might not know if that is what was meant... which is why `JsonWebKey2020` needs to be more specific.

Reply:
> Why not? You have all the information you need, don't you? It's meant to be a public key -- it says it right there. Why are you not throwing an error when you see `d` in a public key!? :)

because https://github.com/w3c/did-spec-registries/blob/master/contexts/did-v1.jsonld#L63

I am relying on JSON development conventions... I'm not against defining all of it again in did core :)

But, it seems a terrible waste of time to copy all the terms and definitions from:

- https://tools.ietf.org/html/rfc7517
- https://www.iana.org/assignments/jose/jose.xhtml#web-key-parameters

I was thinking that it would easier to keep key representation details in the Linked Data Signature Suites, which define the verification method type, and keep did core to general support for widely used cryptography. 

I assume we don't want to / cannot link directly to the w3c ccg work items / linked data signature suites, but I do clearly explain this to readers: https://w3c-ccg.github.io/lds-jws2020/#json-web-key-2020

I'm hearing a contradiction in the key representations topics on my open PRs, perhaps we need another special topic call on key representations.

I'm hearing "define this more in did core" but also "we don't want to define linked data verification methods in did core"... given that, Im not sure how I can help progress this much more than I am now with the PRs I have open.

Reply:
> I was thinking that it would be easier to keep key representation details in the Linked Data Signature Suites, which define the verification method type, and keep did core to general support for widely used cryptography.

Agreed.

> I assume we don't want to / cannot link directly to the w3c ccg work items / linked data signature suites, but I do clearly explain this to readers

We do and can, see the NOTE at the very bottom of this section:

https://www.w3.org/TR/vc-data-model/#proofs-signatures

That section is also an example of what I hope we end up with in DID Core -- we talk about the properties and what they can contain, we provide examples, but don't go into details of each cryptographic suite.